### ISSUE TITLE
Merge Raspberry Pi and Pi2 image

### ISSUE THREAD

**Issue by BertLindeman** [ev3dev/ev3dev#335]
Have installed the Pi2 image. Simply a nice piece of cake, even the sd-card image is expanded by default and automatically. I wonder if merging the two Pi images into one image is in the planning.
So it would allow one to have on sd-card for both machines. If I am correct the raspbian image is compatible with Pi A B and Pi2B. Ah, I do not (yet?) have a BrickPi :disappointed:

**Comment by dlech** [comment]
No, I was not planning on doing this. The Raspberry Pi 2 processor uses the armv7 instruction set, so the image for it uses the official Debian armhf repositories, which are compiled for armv7.

The processor in Raspberry Pi (1), on the other hand, uses the armv6 instruction set, so the official Debian armhf won't run on it (you get illegal instructions). And while the official Debian armel (armv4) does run, it takes a performance hit, mostly noticeable in multimedia applications. This is why Raspbian exists. It is a 3rd party build of Debian for armv6 (confusingly also referred to as armfh). So, we are using Raspbian for the `rpi-` ev3dev image.

It is of course possible to run the Raspbian packages on Rapsberry Pi 2, but I decided to use official Debian for a couple reasons:

* There is the potential for a slight performance increase.
* There are more packages available. Raspbian has all of main, but only some of contrib and non-free.

Nevertheless, you can easily modify the ev3dev Raspberry Pi (1) image to run on a Raspberry Pi 2. You will have to boot it on a Raspberry Pi (1) first (or use brickstrap shell) and install the `linux-image-ev3dev-rpi2` package. `flash-kernel` will give an error that the kernel is the wrong "flavor". You will have to run `flash-kernel --machine "Raspberry Pi 2 Model B"` to get the kernel to install. Once you do this, the SD card should also boot on Raspberry Pi 2. You will always have to keep an eye on kernel updates though and you will have to manually install updates by running `flash-kernel --force <kernel-version>`.
